test/unix/test_lwt_bytes.ml

```diff
@@ -677,4 +713,92 @@ let suite = suite "lwt_bytes" [
       let check = "abcdef" = Lwt_bytes.to_string buffer in
       Lwt.return check
     end;
+
+    test "map_file" begin fun () ->
+      let test_file = "bytes_io_data" in
+      let fd = Unix.openfile test_file [O_RDONLY] 0 in
+      let shared = false in
+      let size = 6 in
+      let buffer = Lwt_bytes.map_file ~fd ~shared ~size () in
+      let check = "abcdef" = Lwt_bytes.to_string buffer in
+      let () = Unix.close fd in
+      Lwt.return check
+    end;
+
+    test "page_size" begin fun () ->
+      let size = 4096 in
+      Lwt.return (size = Lwt_bytes.page_size)
+    end;
+
+    test "mincore buffer length = page_size * 2, n_states = 1"
+      ~only_if:(fun () -> not Sys.win32) begin fun () ->
+      test_mincore (Lwt_bytes.page_size * 2) Lwt_bytes.page_size 1
+      >>= fun states ->
+      Lwt.return (states.(0) = true)
+    end;
+
+    test "mincore buffer length = page_size * 2, n_states = 2"
+      ~only_if:(fun () -> not Sys.win32) begin fun () ->
+      Lwt.catch
+        (fun () ->
+        test_mincore (Lwt_bytes.page_size * 2) Lwt_bytes.page_size 2
+        >>= fun states ->
+        Lwt.return (states.(0) = false)
+        )
+        (function
+      | Invalid_argument _message -> Lwt.return true
+      | _ -> Lwt.return false
+        )
+    end;
+
+    test "mincore buffer length = page_size * 2 + 1, n_states = 2"
+      ~only_if:(fun () -> not Sys.win32) begin fun () ->
+      test_mincore (Lwt_bytes.page_size * 2 + 1) Lwt_bytes.page_size 2
+      >>= fun states ->
+      Lwt.return (states.(0) = true && states.(0) = true)
```

Comment:
Same note about brittleness. Compare to the next test, which ignores `states` completely. In fact, it looks like we never want to check `_states` in these tests. The same comment as for `wait_mincore` applies. Checking `states` is only useful if we can force one of the `states` to reliably initially be `false`, then touch the page, and then we can expect it to reliably be `true`. We can then check these conditions.

If we can't have that, we should probably have `test_mincore` do `Lwt.return ()` instead of `Lwt.return states`. Alternatively, it should touch the pages (read them), to make sure that we can legitimately expect `true`, and then we check for `true`. I suggest also to move the comment to above both helper functions, explaining that we were only able to test bounds checking.

Reply:
I don't think that we have the possibility to force the the states to be false, I checked the Unix module and I did not find any useful functions that could help me to do it.

Reply:
I imagine opening a large-enough file will make it `false` fairly reliably, but we probably don't want to do that.